internal/transactionsubmission/engine/signing/signature_service.go

```diff
@@ -0,0 +1,108 @@
+package signing
+
+import (
+	"fmt"
+
+	"golang.org/x/exp/slices"
+
+	"github.com/stellar/stellar-disbursement-platform-backend/db"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/transactionsubmission/engine/preconditions"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/transactionsubmission/utils"
+)
+
+type SignatureService struct {
+	ChAccountSigner   SignatureClient
+	DistAccountSigner SignatureClient
+	HostAccountSigner SignatureClient
+	DistributionAccountResolver
+}
+
+var _ DistributionAccountResolver = (*SignatureService)(nil)
+
+func (s *SignatureService) Validate() error {
+	if s.ChAccountSigner == nil {
+		return fmt.Errorf("channel account signer cannot be nil")
+	}
+
+	if s.DistAccountSigner == nil {
+		return fmt.Errorf("distribution account signer cannot be nil")
+	}
+
+	if s.HostAccountSigner == nil {
+		return fmt.Errorf("host account signer cannot be nil")
+	}
+
+	if s.ChAccountSigner.NetworkPassphrase() != s.DistAccountSigner.NetworkPassphrase() || s.DistAccountSigner.NetworkPassphrase() != s.HostAccountSigner.NetworkPassphrase() {
+		return fmt.Errorf("network passphrase of all signers should be the same")
+	}
+
+	if s.DistributionAccountResolver == nil {
+		return fmt.Errorf("distribution account resolver cannot be nil")
+	}
+
+	return nil
+}
+
+type SignatureServiceOptions struct {
+	// Shared:
+	NetworkPassphrase string
+
+	// DistributionAccount:
+	DistributionPrivateKey string
+	// DistributionAccountEnv:
+	DistributionSignerType SignatureClientType
+
+	// ChannelAccountDB:
+	DBConnectionPool     db.DBConnectionPool
+	EncryptionPassphrase string
+	LedgerNumberTracker  preconditions.LedgerNumberTracker
+	Encrypter            utils.PrivateKeyEncrypter
+}
+
+// NewSignatureService creates a new signature service instance, given the distribution signer type and the options.
+func NewSignatureService(opts SignatureServiceOptions) (SignatureService, error) {
+	distSignerType := opts.DistributionSignerType
+	if !slices.Contains(SignatureClientType("").AllDistribution(), distSignerType) {
+		return SignatureService{}, fmt.Errorf("invalid distribution signer type %q", distSignerType)
+	}
+
+	sigClientOpts := SignatureClientOptions{
+		Type:                   distSignerType,
+		NetworkPassphrase:      opts.NetworkPassphrase,
+		DistributionPrivateKey: opts.DistributionPrivateKey,
+		DBConnectionPool:       opts.DBConnectionPool,
+		EncryptionPassphrase:   opts.EncryptionPassphrase,
+		Encrypter:              opts.Encrypter,
+		LedgerNumberTracker:    opts.LedgerNumberTracker,
+	}
+
+	sigClientOpts.Type = SignatureClientTypeChannelAccountDB
+	chAccountSigner, err := NewSignatureClient(sigClientOpts)
```

Comment:
100% a nit: I think it would be better to break out the `SignatureClientType` into its own parameter and separate it from the `sigClientOpts` for clarity.  Doesn't have to be done now (or ever) as this is fine. 
```golang
chAccountSigner, err := NewSignatureClient(SignatureClientTypeChannelAccountDB, sigClientOpts)
```

Reply:
Done in https://github.com/stellar/stellar-disbursement-platform-backend/pull/174/commits/fb595e586dff2024e8d6f5e6af691dd07b9f6d2f.